--- ./llvmgcc42-2336.11/gcc/fortran/gfortranspec.c.orig	2007-07-11 22:32:42.000000000 +0100
+++ ./llvmgcc42-2336.11/gcc/fortran/gfortranspec.c	2012-09-12 11:01:00.000000000 +0100
@@ -347,7 +347,7 @@
 	  printf ("GNU Fortran (GCC) %s\n", version_string);
 	  printf ("Copyright %s 2007 Free Software Foundation, Inc.\n\n",
 	          _("(C)"));
-	  printf (_("GNU Fortran comes with NO WARRANTY, to the extent permitted by law.\n\
+	  printf ("%s",_("GNU Fortran comes with NO WARRANTY, to the extent permitted by law.\n\
 You may redistribute copies of GNU Fortran\n\
 under the terms of the GNU General Public License.\n\
 For more information about these matters, see the file named COPYING\n\n"));
@@ -525,7 +525,7 @@
 
   if (verbose && g77_newargv != g77_xargv)
     {
-      fprintf (stderr, _("Driving:"));
+      fprintf (stderr, "%s", _("Driving:"));
       for (i = 0; i < g77_newargc; i++)
 	fprintf (stderr, " %s", g77_newargv[i]);
       fprintf (stderr, "\n");
